encoder.go

```diff
@@ -54,13 +68,27 @@ func (encoder *encoder) Encode(data any) (*wafObject, error) {
 	value := reflect.ValueOf(data)
 	wo := &wafObject{}
 
+	defer func() {
+		if encoder.truncations[ObjectTooDeep] != nil {
+			encoder.measureObjectDepth(value)
+		}
+	}()
+
 	if err := encoder.encode(value, wo, encoder.objectMaxDepth); err != nil {
 		return nil, err
 	}
 
 	return wo, nil
 }
 
+// Truncations returns all truncations that happened since the last call to `Truncations()`, and clears the internal
+// list. This is a map from truncation reason to the list of un-truncated value sizes.
+func (encoder *encoder) Truncations() map[TruncationReason][]int {
+	result := encoder.truncations
+	encoder.truncations = nil
```

Comment:
update: I don't see where this function is used (found only one use in tests)

Reply:
The reason it behaves like that is to allow a given `*encoder` instance to be re-used. It feels to me that they can be re-used, as otherwise there would be no use for `Encode` to be a method (it would be a plain function).

The clearing of the set allows you to make sure you only get truncation data for the value you _just_ encoded.

Reply:
Let's talk more about in another PR when you actually use this function